pkg/querier/queryrange/queryrangebase/results_cache.go

```diff
@@ -580,7 +580,7 @@ func (s resultsCache) partition(req Request, extents []Extent) ([]Request, []Res
 }
 
 func (s resultsCache) filterRecentExtents(req Request, maxCacheFreshness time.Duration, extents []Extent) ([]Extent, error) {
-	maxCacheTime := (int64(model.Now().Add(-maxCacheFreshness)) / req.GetStep()) * req.GetStep()
```

Comment:
IIUC, here the multiply cancels the division. Also, the step for an index stat req is 0 so this yields a divide by 0 panic.

Reply:
It's important to leave this for other types as this aligns the cache time with a multiple of the `step` via integer division. This keeps entries in the cache identical to the steps used in the query. This is important for metric queries. Instead, if the step is zero, can we consider it `1`?